src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -99,6 +99,15 @@ internal Data(int trackedEntries, bool requiredAnalysis)
             public readonly TypeWithState RValueType;
             public readonly TypeWithAnnotations LValueType;
 
+            // For lambda expressions, we save the state when the lambda is analyzed,
+            // so we can use it when analyzing the lambda body as part of a delegate conversion.
+            public readonly Optional<LocalState> StateForLambda;
+
+            // For expressions that contain nested expressions (such as collection expressions),
+            // we store the results of visiting those nested expressions.
+            // Note: we cannot use ImmutableArray<VisitResult> because that yields a TypeLoad exception on .NET Framework.
```

Comment:
Ughhh it thinks it's a layout cycle doesn't it.

Reply:
Yes, I think so